src/ImageSharp/Formats/Jpeg/JpegMetadata.cs

```diff
@@ -21,18 +27,94 @@ public JpegMetadata()
         {
-            this.Quality = other.Quality;
             this.ColorType = other.ColorType;
+
+            this.LuminanceQuantizationTable = other.LuminanceQuantizationTable;
+            this.ChromaQuantizationTable = other.ChromaQuantizationTable;
+            this.LuminanceQuality = other.LuminanceQuality;
+            this.ChrominanceQuality = other.ChrominanceQuality;
         }
 
         /// <summary>
-        /// Gets or sets the encoded quality.
+        /// Gets or sets luminance qunatization table for jpeg image.
         /// </summary>
-        public int Quality { get; set; } = 75;
+        internal Block8x8F LuminanceQuantizationTable
+        {
+            get
+            {
+                if (this.lumaQuantTable.HasValue)
+                {
+                    return this.lumaQuantTable.Value;
+                }
+
+                return Quantization.ScaleLuminanceTable(this.LuminanceQuality ?? Quantization.DefaultQualityFactor);
+            }
+
+            set => this.lumaQuantTable = value;
+        }
+
+        /// <summary>
+        /// Gets or sets chrominance qunatization table for jpeg image.
+        /// </summary>
+        internal Block8x8F ChromaQuantizationTable
+        {
+            get
+            {
+                if (this.chromaQuantTable.HasValue)
+                {
+                    return this.chromaQuantTable.Value;
+                }
+
+                return Quantization.ScaleChrominanceTable(this.ChrominanceQuality ?? Quantization.DefaultQualityFactor);
+            }
+
+            set => this.chromaQuantTable = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the jpeg luminance quality.
+        /// </summary>
+        /// <remarks>
+        /// This value might not be accurate if it was calculated during jpeg decoding
+        /// with non-complient ITU quantization tables.
+        /// </remarks>
+        public int? LuminanceQuality { get; set; }
+
+        /// <summary>
+        /// Gets or sets the jpeg chrominance quality.
+        /// </summary>
+        /// <remarks>
+        /// This value might not be accurate if it was calculated during jpeg decoding
+        /// with non-complient ITU quantization tables.
+        /// </remarks>
+        public int? ChrominanceQuality { get; set; }
 
         /// <summary>
         /// Gets or sets the encoded quality.
         /// </summary>
+        /// <remarks>
+        /// Note that jpeg image can have different quality for luminance and chrominance components.
+        /// This property return average for both qualities and sets both qualities to the given value.
+        /// </remarks>
+        public int Quality
+        {
+            get
+            {
+                int lumaQuality = this.LuminanceQuality ?? Quantization.DefaultQualityFactor;
+                int chromaQuality = this.ChrominanceQuality ?? lumaQuality;
+                return (int)Math.Round((lumaQuality + chromaQuality) / 2f);
```

Comment:
Does JPEGsnoop also report the average of the two values in `Approx quality factor`?

Reply:
JPEGsnoop has no 'quality' parameter, only separate luma/chroma. That's the main pain in the neck to define `Quality` alongside luma/chroma separation.

Reply:
It shouldn't be that bad for general images which have same quality levels for both tables. Jpegs with different quality levels for luma/chroma was a surprise to me and shouldn't be really popular.

Reply:
Are there any other libraries/tools which report a quality estimation after import?

Reply:
[Forensically](https://29a.ch/photo-forensics/#forensic-magnifier)
General quality
On different `luma(71.19)`/`chroma(80.24)` quality it returns `quality=81` - `ceil(max(luma, chroma))`?.

JPEGsnoop
Separate luma/chroma qualities
No general quality

ImageMagick
General quality

That's all I know, photoshop or any other tools for image manipulations do not expose such info afaik.
I guess not changing anything in public interface is the solution. We can always expose it later as internally metadata would still store separate qualities.

The only question is:
1. `quality => avg(luma, chroma)`
2. `quality => ceil(max(luma, chroma))`

This public property won't affect encoding process anyway but for user IMO it's better to expose max quality factor than average for reasons I stated in the comments above (loss of detail + extra noise and filesize).

Reply:
Great research, thanks! I also think that taking the maximum would be better.